samples/dnn/text_detection.cpp

```diff
@@ -20,65 +20,104 @@
+    if (argc == 1 || parser.has("help")) {
         parser.printMessage();
         return 0;
     }
 
-    float confThreshold = parser.get<float>("thr");
-    float nmsThreshold = parser.get<float>("nms");
-    int width = parser.get<int>("width");
-    int height = parser.get<int>("height");
+    // Parsing command-line arguments
+    String detModelPath = parser.get<String>("detModelPath");
+    String detModelConfig = parser.get<String>("detModelConfig");
+    String recModelPath = parser.get<String>("recModelPath");
+    int height = parser.get<int>("inputHeight");
+    int width = parser.get<int>("inputWidth");
     int imreadRGB = parser.get<int>("RGBInput");
-    String detModelPath = parser.get<String>("detModel");
-    String recModelPath = parser.get<String>("recModel");
     String vocPath = parser.get<String>("vocabularyPath");
+    float binThresh = parser.get<float>("binaryThreshold");
+    float polyThresh = parser.get<float>("polygonThreshold");
+    double unclipRatio = parser.get<double>("unclipRatio");
+    uint maxCandidates = parser.get<uint>("maxCandidate");
+    float confThreshold = parser.get<float>("thr");
+    float nmsThreshold = parser.get<float>("nms");
 
-    if (!parser.check())
-    {
+    // Ensuring the provided arguments are valid
+    if (!parser.check()) {
         parser.printErrors();
         return 1;
     }
 
-    // Load networks.
-    CV_Assert(!detModelPath.empty() && !recModelPath.empty());
-    TextDetectionModel_EAST detector(detModelPath);
-    detector.setConfidenceThreshold(confThreshold)
-            .setNMSThreshold(nmsThreshold);
-
-    TextRecognitionModel recognizer(recModelPath);
+    // Asserting detection model path is provided
+    CV_Assert(!detModelPath.empty());
+
+    std::vector<std::vector<Point>> detResults;
+    // Reading the input image
+    Mat frame = imread(samples::findFile(parser.get<String>("input")));
+
+    // Initializing and configuring the text detection model based on the provided config
+    if (detModelConfig == "east") {
+        // EAST Detector initialization
+        TextDetectionModel_EAST detector(detModelPath);
+        detector.setConfidenceThreshold(confThreshold)
+                .setNMSThreshold(nmsThreshold);
+        // Setting input parameters specific to EAST model
+        detector.setInputParams(1.0, Size(width, height), Scalar(123.68, 116.78, 103.94), true);
+        // Performing text detection
+        detector.detect(frame, detResults);
+    }
+    else if (detModelConfig == "db") {
+        // DB Detector initialization
+        TextDetectionModel_DB detector(detModelPath);
+        detector.setBinaryThreshold(binThresh)
+                .setPolygonThreshold(polyThresh)
+                .setUnclipRatio(unclipRatio)
+                .setMaxCandidates(maxCandidates);
+        // Setting input parameters specific to DB model
+        detector.setInputParams(1.0 / 255.0, Size(width, height), Scalar(122.67891434, 116.66876762, 104.00698793));
```

Comment:
Same request as for EAST. Let's move magic constants to `TextDetectionModel_DB` constructor